### ISSUE TITLE
Option to pull character attributes on ship rolls

### ISSUE THREAD

**Issue by CubicleDexter** [Karlinator/roll20-character-sheets#84]
Would it be possible to add an option where the ship side of the character sheet will use the attributes of the same-sheet character?

Use case: Ship weapon attacks add character attack value and attribute mod to attack, and add attribute mod to damage. Each character could add the ship's weapons to their character's ship sheet so that attack and damage are automatically calculated when they are in the gunner's seat.

**Comment by Karlinator** [comment]
You can already add the ship weapons to your own character sheet in the character weapon section to achieve this.

I have been toying with the idea of specifying a gunner on the ship sheet and pulling stats from there, but that requires the name be spelt exactly right (and as in the journal not the character name), so it's maybe a bit unintuitive to use.

**Comment by CubicleDexter** [comment]
That's true, I forgot about that (and the Discord discussion that I just now remembered also having) for PC characters.

Instead of specifying a gunner on the sheet, is it possible to pull stats from a selected token like Initiative?

**Comment by Karlinator** [comment]
Ooo, that's a good point. I think that is possible (using {Selected|AB} rather than {Character|AB}).

It's still a bit clunky, as it might just break if you don't have a token selected (not sure yet, haven't tested). Either way you'd want a toggle I think of whether to do the fancy shit or just a straight roll.

**Comment by CubicleDexter** [comment]
And then there's the distinction between attacking with Shoot or Pilot. Honestly, ships are a pain in the arse.

Toggle options are always nice on the user side. I have no idea how much extra work they are for dev.

**Comment by Karlinator** [comment]
I just have to write the roll for each option, put it in the value of a checkbox or dropdown or something, then refer to that attribute. Roll20 does actually parse the macro text from inside an attribute call.

Shoot/Pilot is one thing, so is Dex/Int. This is easy when adding it as a character weapon, but you can also just have a dropdown for that (Roll20, I believe, parses nested macros all the way down).
